PR: https://github.com/odoo/odoo/pull/

From: fe631a36a5c7278040875434ad74039dfa1cf92a
From: rap-odoo
Date: 2017-05-15 12:09:11

Structural Changes: 4
Total Changes: 13

[IMP] point_of_sale: allow product grouping for all uom

Purpose
=======

In POS, it is hard-codded in such a way that grouping will be done for the
uom category that has database id == 1.

Specification
=============

We have solved this issue: we have added a new boolean field in the uom category form view that

If a user chooses this option then products having that uom category will be grouped in the POS.

Both uom category Unit and Weight have been set to groupable by default.

- Removed the hardcoded part of groupable for uom which had database id == 1,
  Now grouping will be done on 'groupable' field of "product.uom", which is
  related to the field of the same name on "product.uom.category".
- Only the `Unit` uom category is grouped by default

================================= pseudo patch: =================================

--- a/addons/point_of_sale/models/product.py
+++ b/addons/point_of_sale/models/product.py
@@ -20,3 +20,16 @@ class ProductTemplate(models.Model):
             if self.env['pos.session'].search_count([('state', '!=', 'closed')]):
                 raise UserError(_('You cannot delete a product saleable in point of sale while a session is still opened.'))
         return super(ProductTemplate, self).unlink()
+
+
+class ProductUomCateg(models.Model):
+    _inherit = 'product.uom.categ'
+
+    is_pos_groupable = fields.Boolean(string='Group Products in POS',
+        help="Check if you want to group products of this category in point of sale orders")
+
+
+class ProductUom(models.Model):
+    _inherit = 'product.uom'
+
+    is_pos_groupable = fields.Boolean(related='category_id.is_pos_groupable')
